dali/test/python/test_operator_numba_func.py

```diff
@@ -14,126 +14,162 @@
-
-@cfunc(dali_numba.run_fn_sig(types.uint8, types.uint8), nopython=False)
-def rot_image(out_ptr, out_shape_ptr, ndim_out, in_ptr, in_shape_ptr, ndim_in):
-    out_shape = carray(out_shape_ptr, ndim_out)
-    in_shape = carray(in_shape_ptr, ndim_in)
-    in_arr = carray(in_ptr, (in_shape[0], in_shape[1], in_shape[2]))
-    out_arr = carray(out_ptr, (out_shape[0], out_shape[1], out_shape[2]))
-    for i in range(out_shape[0]):
-        for j in range(out_shape[1]):
-            out_arr[i][j] = in_arr[j][out_shape[0] - i - 1]
-
-@cfunc(dali_numba.setup_fn_sig(1, 1), nopython=True)
-def rot_image_setup(out_shape_ptr, out1_ndim, out_dtype, in_shape_ptr, in1_ndim, in_dtype, num_samples):
-    in_arr = carray(in_shape_ptr, num_samples * out1_ndim)
-    out_arr = carray(out_shape_ptr, num_samples * in1_ndim)
-    out_type = carray(out_dtype, 1)
-    out_type[0] = in_dtype
-    for i in range(0, num_samples * out1_ndim, 3):
-        out_arr[i] = in_arr[i + 1]
-        out_arr[i + 1] = in_arr[i]
-        out_arr[i + 2] = in_arr[i + 2]
+def reverse_col_batch(out0, in0):
+    for sample_id in range(len(out0)):
+        out0[sample_id][:] = 255 - in0[sample_id][:]
+
+def reverse_col_sample(out0, in0):
+    out0[:] = 255 - in0[:]
+
+def rot_image_batch(out0, in0):
+    for out_sample, in_sample in zip(out0, in0):
+        for i in range(out_sample.shape[0]):
+            for j in range(out_sample.shape[1]):
+                out_sample[i][j] = in_sample[j][out_sample.shape[0] - i - 1]
+
+def rot_image_sample(out0, in0):
+    for i in range(out0.shape[0]):
+        for j in range(out0.shape[1]):
+            out0[i][j] = in0[j][out0.shape[0] - i - 1]
+
+def rot_image_setup(outs, ins):
+    out0 = outs[0]
+    in0 = ins[0]
+    for sample_id in range(len(out0)):
+        out0[sample_id][0] = in0[sample_id][1]
+        out0[sample_id][1] = in0[sample_id][0]
+        out0[sample_id][2] = in0[sample_id][2]
 
 def test_numba_func_image():
     args = [
-        (reverse_col.address, None, lambda x: 255 - x),
-        (rot_image.address, rot_image_setup.address, lambda x: np.rot90(x)),
+        (reverse_col_batch, [dali_types.UINT8], [dali_types.UINT8], [3], [3], None, True, lambda x: 255 - x),
+        (reverse_col_sample, [dali_types.UINT8], [dali_types.UINT8], [3], [3], None, None, lambda x: 255 - x),
+        (rot_image_batch, [dali_types.UINT8], [dali_types.UINT8], [3], [3], rot_image_setup, True, lambda x: np.rot90(x)),
+        (rot_image_sample, [dali_types.UINT8], [dali_types.UINT8], [3], [3], rot_image_setup, None, lambda x: np.rot90(x)),
     ]
-    for fn_ptr, setup_fn, transform in args:
-        yield _testimpl_numba_func_image, fn_ptr, setup_fn, transform
+    for run_fn, out_types, in_types, outs_ndim, ins_ndim, setup_fn, batch_processing, transform in args:
+        yield _testimpl_numba_func_image, run_fn, out_types, in_types, outs_ndim, ins_ndim, setup_fn, batch_processing, transform
+
+def split_images_col_sample(out0, out1, out2, in0):
```

Comment:
maybe add another example taking 3 inputs R, G, B and producing a single output with 3 channels